code/object/deadobjectdock.cpp

```diff
@@ -149,6 +144,16 @@ void dead_dock_remove_instance(object *objp, object *other_objp)
 		// delete it
 		vm_free(ptr);
 	}
+	else
+	{
+		// Raise an error if we're in a release build.
+		// Raise a warning if we're in a debug build, the user may be doing something silly
+#ifndef NDEBUG
+		Error(LOCATION, "Tried to undock an object that isn't docked!\n");
```

Comment:
Same as above, this should either always be an error or it should always be a warning. If the warning should appear in release builds then you can use `ReleaseWarning`.